server/src/main/java/org/elasticsearch/action/search/SearchResponseMerger.java

```diff
@@ -178,10 +178,17 @@ SearchResponse getMergedResponse(Clusters clusters) {
                 assert trackTotalHits == null || trackTotalHits;
                 trackTotalHits = true;
             }
-            TopDocs topDocs = searchHitsToTopDocs(searchHits, totalHits, shards);
-            topDocsStats.add(new TopDocsAndMaxScore(topDocs, searchHits.getMaxScore()),
-                searchResponse.isTimedOut(), searchResponse.isTerminatedEarly());
-            topDocsList.add(topDocs);
+            if (totalHits.value > 0) {
```

Comment:
If track_total_hits is disabled you can have totalHits.value equals to 0 even though searchHits are not empty. I think something like this should work:
````
TopDocs topDocs = searchHitsToTopDocs(searchHits, totalHits, shards);
 topDocsStats.add(new TopDocsAndMaxScore(topDocs, searchHits.getMaxScore()),
                searchResponse.isTimedOut(), searchResponse.isTerminatedEarly());
if (searchHits.getHits().length == 0) {
  topDocsList.add(topDocs);
}
````
?

Reply:
I was going to ask you about this, not sure. Let me try it.

Reply:
works, I pushed an update. I also had to adjust topDocsToSearchHits to accept null TopDocs which happens when all clusters return empty search hits.